src/OrchardCore.Modules/OrchardCore.Users/Views/Items/RegisterUserTask.Fields.Edit.cshtml

```diff
@@ -11,14 +11,27 @@
 
 <div class="form-group" asp-validation-class-for="ConfirmationEmailSubject">
     <label asp-for="ConfirmationEmailSubject">@T["Subject for confirmation email"]</label>
-    <input type="text" asp-for="ConfirmationEmailSubject" class="form-control code" />
+    <input type="text" asp-for="ConfirmationEmailSubject" class="form-control code" required />
     <span asp-validation-for="ConfirmationEmailSubject"></span>
     <span class="hint">@T["The subject of the message. Supports Liquid syntax."]</span>
 </div>
 
 <div class="form-group" asp-validation-class-for="ConfirmationEmailTemplate">
     <label asp-for="ConfirmationEmailTemplate">@T["Template for confirmation email"]</label>
-    <input type="text" asp-for="ConfirmationEmailTemplate" class="form-control code" />
+    <input type="text" asp-for="ConfirmationEmailTemplate" class="form-control code" required />
     <span asp-validation-for="ConfirmationEmailTemplate"></span>
     <span class="hint">@T["The message to send. Access email confirmation url with Workflow.Properties.EmailConfirmationUrl liquid property."]</span>
 </div>
+
+<script at="Foot">
+    $(function () {
+        var required = false;
+
+        $("#@Html.IdFor(m => m.SendConfirmationEmail)").change(function () {
+            $("#@Html.IdFor(m => m.ConfirmationEmailSubject)").attr('required', required);
```

Comment:
Fine to add some javascript helpers, but the validation also needs to be performed in the display driver

Reply:
I will try to add some validation on the display driver, coz all the time I see the validation on view model using data annotations

Reply:
@deanmarcussen I added a validation on the display driver, but I think it's not needed because the workflow already checked the `SendConfirmationEmail` before set the expressions for other properties